mypy/checker.py

```diff
@@ -3725,7 +3725,7 @@ def make_fake_typeinfo(self,
         return cdef, info
 
     def intersect_instances(self,
-                            instances: Sequence[Instance],
+                            instances: Tuple[Instance, Instance],
```

Comment:
Why do you need this change? I don't see anything below that requires there to be exactly two.

Reply:
My change covers two permutations of the given `Instance` objects: it checks the original and the reversed order.  For the given order `(a, b, c)`, we would also need to check the order `(b, a, c)`, for example.

Reply:
Just a reminder. Or can I help with further explanations?